File: 2024/listas_dicionarios_hashs/Aluno.java
public class Aluno {
    public int matricula;
    public String nome;
    public String email;

    public Aluno(int matricula, String nome) {
        this.matricula = matricula;
        this.nome = nome;
        this.email = this.gerarEmail();
    }

    public String gerarEmail() {
        String vetorNomes[];
        vetorNomes = this.nome.toLowerCase().split(" ");
        return vetorNomes[0] + "." + vetorNomes[vetorNomes.length - 1] + "@ufn.edu.br";
    }

    @Override
    public String toString() {
        return "Aluno [nome=" + nome + ", email=" + email + "]";
    }

    @Override
    public boolean equals(Object obj) {       
        Aluno aluno = (Aluno) obj;        
        if (this.email.equals(aluno.email))
            return true;
        return false;
    }
    
    
}

File: 2024/listas_dicionarios_hashs/aluno.py
class Aluno:
    #construtor
    def __init__(self, matricula, nome):
        self.matricula = matricula
        self.nome = nome
        self.email = self.gerar_email()
    
    def gerar_email(self):
        vetor_nomes = self.nome.lower().split(" ")
        return vetor_nomes[0] + "." + vetor_nomes[-1] + "@ufn.edu.br"
    
    def __str__(self):
        return "Aluno [nome=" + self.nome + ", email=" + self.email + "]"

    def __eq__(self, obj):
        # Verificando se obj é uma instância da classe Aluno
        if not isinstance(obj, Aluno):
            return False
        # Comparando o email dos dois objetos
        return self.email == obj.email


